internal/ldconfig/ldconfig.go

```diff
@@ -130,7 +130,8 @@ func (l *Ldconfig) prepareRoot() (string, error) {
 	// To prevent leaking the parent proc filesystem, we create a new proc mount
 	// in the specified root.
 	if err := mountProc(l.inRoot); err != nil {
-		return "", fmt.Errorf("error mounting /proc: %w", err)
+		// Non-root users cannot mount /proc; print the error but continue execution
```

Comment:
Is there a way to detect whether this is running as a non-root user instead?

Reply:
I tried some ways.

At first, I thought getting uid and gid would help. But it outcomes "0" in the containers, caused by sandboxed namespaces.

AFAIK, the only way to test mounting /proc is to trying it.

Reply:
I agree. I believe we can avoid side effects (e.g., unexpected behavior for root users) by adding a third argument `rootless=true` to both `soname-symlinks` and `update-ldcache`. I've rebased the code based on your suggestion, with the following additional changes:

1. Optional argument: In your suggested code, I made the `rootless=` argument optional to ensure it does not affect existing `nvidia-ctk` behavior for end users. This helps avoid any potential side effects.
1. Added `remainingArgs`: The new `NewFromArgs` function now returns the remaining arguments it did not consume. This allows `soname-symlinks` and `update-ldcache` to forward any remaining arguments directly to ldconfig.

Reply:
Thanks for confirming that this works. 

Note that I have created https://github.com/elezar/nvidia-container-toolkit/tree/fix/suppress-proc-mount-error-for-non-root-users that takes advantage of a refactor from #1174 to improve how the arguments are handled.

One question that I also have is whether the pivot_root operation works as expected in a rootless container, or whether we would have to fallback to a "standard" chroot?